### PR TITLE
feat: Allow SchemaForm to be form or div

### PR DESCRIPTION
**Pull request by marina-mosti**

### CODE REVIEW

File: .vitepress/docs/components/SchemaPlayground.vue [hunk 1]
```diff
@@ -10,9 +10,14 @@
     </div>
     <div>
       <SchemaForm
+        @submit.prevent="onSubmit"
```
Comment (shentao) [inline .vitepress/docs/components/SchemaPlayground.vue]: I think we can remove the `.prevent` here as it’s not necessary, right?

File: src/SchemaForm.vue [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
-        v-bind="binds(field)"
-        :modelValue="val(field)"
-        @update:modelValue="update(field.model, $event)"
-        @update-batch="updateBatch(field.model, $event)"
-      />
-      <slot name="afterForm"></slot>
-    </div>
+    <component
+      :is="!hasParentSchema ? 'form' : 'div'"
+      class="schema-form"
```
Comment (shentao) [inline src/SchemaForm.vue]: What does the `class` do here?

**Review by shentao — APPROVED** [review]

File: .vitepress/docs/components/SchemaPlayground.vue (hunk 1 above)
Comment (marina-mosti) [reply .vitepress/docs/components/SchemaPlayground.vue]: It actually is, when not present the page reloads itself

File: src/SchemaForm.vue (hunk 2 above)
Comment (marina-mosti) [reply src/SchemaForm.vue]: Its being used by parents to style, and i think users may want to use it to target this particular form in style sheets

File: .vitepress/docs/components/SchemaPlayground.vue (hunk 1 above)
Comment (shentao) [reply .vitepress/docs/components/SchemaPlayground.vue]: Oh I see! That makes sense! I think we need to add that listener with `.prevent` to that root `<form>` element in SchemaForm.vue.

File: src/SchemaForm.vue (hunk 2 above)
Comment (shentao) [reply src/SchemaForm.vue]: Hm. Given that `class` and `style` are being attached to the root element, do we really need it?

Comment (marina-mosti) [reply src/SchemaForm.vue]: Alright sounds good

File: src/SchemaForm.vue [hunk 3]
```diff
@@ -1,23 +1,27 @@
...
-        :modelValue="val(field)"
-        @update:modelValue="update(field.model, $event)"
-        @update-batch="updateBatch(field.model, $event)"
-      />
-      <slot name="afterForm"></slot>
-    </div>
+    <component
+      :is="!hasParentSchema ? 'form' : 'div'"
+      v-bind="formBinds"
+      >
```
Comment (shentao) [inline src/SchemaForm.vue]: Unnecessary extra indentation.

**Review by shentao — APPROVED** [review]

**Pull request merged**